Communications/Constants.cs

```diff
@@ -1,5 +1,23 @@
 ﻿namespace Scarlet.Communications
 {
+    /// <summary> Represents data type. </summary>
+    public enum TypeID
+    {
+        BOOL = 0,
+        CHAR = 1,
+        DOUBLE = 2,
+        FLOAT = 3,
+        INT = 4,
+        LONG = 5,
+        SHORT = 6,
+        UINT = 7,
+        ULONG = 8,
+        USHORT = 9,
+        BYTE = 10,
+        STRING = 11,
+        BYTES = 12,
+    }
```

Comment:
I don't think it's necessary to define the integer value of each Enum constant.

Reply:
The reason I hard coded the value for each constant is to avoid accidentally causing the type ID of following types to change after removing a type from the list, which will break the compatibility between different versions. It will also make it more easy to lookup what an ID stands for when debugging (for example, understanding the packet captured by Wireshark).

Reply:
Is there a reason that you are not using the .NET built-in `TypeCode` instead?

Reply:
Well, I didn't know that... I think that is a good idea, except a small problem that there's not `TypeID` for `byte[]`.

Reply:
Given that `TypeCode` doesn't have a `byte[]` element and I don't see a better way to do this with just using the `Type` class, I think this is an okay solution.